app/src/main/java/io/homeassistant/companion/android/sensors/PhoneStateSensorManager.kt

```diff
@@ -124,4 +143,20 @@ class PhoneStateSensorManager : SensorManager {
             )
         }
     }
+    fun updateCallNumber(context: Context, intent: Intent) {
+        if (!isEnabled(context, callNumber.id))
+            return
+        if (checkPermission(context)) {
+            if (intent.hasExtra(TelephonyManager.EXTRA_INCOMING_NUMBER)) {
+                val number = intent.getStringExtra(TelephonyManager.EXTRA_INCOMING_NUMBER)
```

Comment:
What is the default state if we can't retrieve the number? Or if there is no active call?

Reply:
There is 2 ways for it:
1. Call is finished. In this case phoneState sensor changes to "idle" and function updatePhoneStateSensor(context) will update this sensor with "none" state.
2. We don't have READ_CALL_LOG permission, sosensor willbe disabled.

The reason of this strange logic is that when we have READ_CALL_LOG permission, system send ACTION_PHONE_STATE_CHANGED broadcast twice. But only one time with EXTRA_INCOMING_NUMBER added.

Reply:
So based on this, I wonder if we should consider hiding this sensor until phone state sensor is enabled? It sounds like its more of a requirement right?  Like the user needs to first grant access to even detect if the phone state has changed, then they can get the next call number.

Reply:
Actually it will not do anything without PhoneState sensor enabled. We skip sensors update in sensor receiver before this function can get this intent.

Reply:
Right but its a better user experience to hide and disable a sensor they can't use if it depends on another sensor being enabled.  It will confuse a user if they can see the sensor and attempt to enable it just for it not to update.

Reply:
Ok, I believe your expirience. I wounder how do you get time and patient to troubleshout all of this complaints on community forum, reddit, github and so on. Nice work, mate! 👍

Reply:
Haha the flow is generally the same at my job and working from home makes it easy to be responsive :) Thank you